packages/@lwc/engine-core/src/renderer.ts

```diff
@@ -0,0 +1,283 @@
+    getElementsByClassName = getElementsByClassNameImpl;
+}
+
+type getChildrenFunc = (element: E) => HTMLCollection;
+export let getChildren: getChildrenFunc;
+export function setGetChildren(getChildrenImpl: getChildrenFunc) {
+    getChildren = getChildrenImpl;
+}
+
+type getChildNodesFunc = (element: E) => NodeList;
+export let getChildNodes: getChildNodesFunc;
+export function setGetChildNodes(getChildNodesImpl: getChildNodesFunc) {
+    getChildNodes = getChildNodesImpl;
+}
+
+type getFirstChildFunc = (element: E) => N | null;
+export let getFirstChild: getFirstChildFunc;
+export function setGetFirstChild(getFirstChildImpl: getFirstChildFunc) {
+    getFirstChild = getFirstChildImpl;
+}
+
+type getFirstElementChildFunc = (element: E) => E | null;
+export let getFirstElementChild: getFirstElementChildFunc;
+export function setGetFirstElementChild(getFirstElementChildImpl: getFirstElementChildFunc) {
+    getFirstElementChild = getFirstElementChildImpl;
+}
+
+type getLastChildFunc = (element: E) => N | null;
+export let getLastChild: getLastChildFunc;
+export function setGetLastChild(getLastChildImpl: getLastChildFunc) {
+    getLastChild = getLastChildImpl;
+}
+
+type getLastElementChildFunc = (element: E) => E | null;
+export let getLastElementChild: getLastElementChildFunc;
+export function setGetLastElementChild(getLastElementChildImpl: getLastElementChildFunc) {
+    getLastElementChild = getLastElementChildImpl;
+}
+
+type isConnectedFunc = (node: N) => boolean;
+export let isConnected: isConnectedFunc;
+export function setIsConnected(isConnectedImpl: isConnectedFunc) {
+    isConnected = isConnectedImpl;
+}
+
+type insertGlobalStylesheetFunc = (content: string) => void;
+export let insertGlobalStylesheet: insertGlobalStylesheetFunc;
+export function setInsertGlobalStylesheet(insertGlobalStylesheetImpl: insertGlobalStylesheetFunc) {
+    insertGlobalStylesheet = insertGlobalStylesheetImpl;
+}
+
+type insertStylesheetFunc = (content: string, target: N) => void;
+export let insertStylesheet: insertStylesheetFunc;
+export function setInsertStylesheet(insertStylesheetImpl: insertStylesheetFunc) {
+    insertStylesheet = insertStylesheetImpl;
+}
+
+type assertInstanceOfHTMLElementFunc = (elm: any, msg: string) => void;
+export let assertInstanceOfHTMLElement: assertInstanceOfHTMLElementFunc;
+export function setAssertInstanceOfHTMLElement(
+    assertInstanceOfHTMLElementImpl: assertInstanceOfHTMLElementFunc
+) {
+    assertInstanceOfHTMLElement = assertInstanceOfHTMLElementImpl;
+}
+
+type defineCustomElementFunc = (
+    name: string,
+    constructor: CustomElementConstructor,
+    options?: ElementDefinitionOptions
+) => void;
+export let defineCustomElement: defineCustomElementFunc;
+export function setDefineCustomElement(defineCustomElementImpl: defineCustomElementFunc) {
+    defineCustomElement = defineCustomElementImpl;
+}
+
+type getCustomElementFunc = (name: string) => CustomElementConstructor | undefined;
```

Comment:
not sure why it is `| undefined`, some weird stuff related to SSR?

Reply:
This was the pre-existing type:

https://github.com/salesforce/lwc/blob/1b2e246e03c6bfbc80bea3def43b7d0281a1de21/packages/%40lwc/engine-core/src/framework/renderer.ts#L64

I think you're right though that we don't need the `| undefined`. Can we address in another PR? I tried to avoid changing any of these types.